src/datagen.py

```diff
@@ -196,6 +196,7 @@ def __init__(
         )
 
         self.weather_features_df = self.init_weather_features()
+        self.weather_features_matrix = self.weather_features_df.sort_values(by = 'weather_file_city')[self.weather_features].to_numpy()
```

Comment:
Sorting values alphabetically by the weather file city should ensure that the row numbers correspond to the same index output by the `StringIndexer` which assigns the indices in alphabetical order

Reply:
I think we have to be very careful about sorting in two different places and hoping for the best. I would be much happier if there were a single persisted mapping between city names and indices that was used at both training and inference time and tied to a fitted model version. If you mess up the mapping for any reason then you start serving garbage predictions and there is absolutely nothing to indicate that this is happening. You just used weather index 47 instead of 137 and got the wrong answer. I'd really rather not have any "should" in the approach we take to this.

Reply:
OK, I think that makes sense, @mikivee. I guess the final step is that when we serve this to PEP or the world at large we have to make sure that we load the exact right version of the two columns for the weather city and corresponding index and when the caller gives us an address, we geocode to a weather city, look up the index, and pass that to the model for inference.